### PR TITLE
Update to 1.8

### PR DESCRIPTION
**Pull request by Bart-0110**

### CODE REVIEW

**Comment by Bart-0110** [comment]
Wait..
Don't do it yet. It will replace the translation with hard coded strings and it will disable the updater

**Comment by Bart-0110** [comment]
The translation works again. But I don't know how the updater works, but I uncomment the updater part so you can merge it :)

**Comment by licks92** [comment]
Oh wait dude. The translation system works just fine. But the step used to make it work requires to get the strings from crowdin.

Well that's complicated, but that's what allows me to translate it in... I don't remember how many languages but that's a lot.

I will explain to you later. For now, I'm looking at the code. eventually i will merge it, and then i will fix translation (should be quick)

**Comment by Bart-0110** [comment]
The problem with translation was that the variables were not initialized. If you use a command for example you get a nullpointerexception

File: src/main/java/net/licks92/WirelessRedstone/Strings/WirelessXMLStringsLoader.java [hunk 1]
```diff
@@ -2,123 +2,234 @@
...
-{
+public class WirelessXMLStringsLoader extends WirelessStrings {
 	private final String STRINGS_FOLDER = "languages/";
 	private final String defaultLanguage = "en";
-	
-	public WirelessXMLStringsLoader(WirelessRedstone plugin, String language)
-	{
-		String defaultLocation = STRINGS_FOLDER + defaultLanguage + "/strings.xml";
-		InputStream stream = plugin.getResource(STRINGS_FOLDER + language + "/strings.xml");
-		if(stream != null) {
```
Comment (licks92) [inline src/main/java/net/licks92/WirelessRedstone/Strings/WirelessXMLStringsLoader.java]: So these are the 3 lines that were removed?
That's okay i will reimplement this and explain it in comments ;)

**Comment by licks92** [comment]
Ikr.
Let me explain...

**Comment by licks92** [comment]
The principle is that the strings.xml files were not on your computer.
And that's normal.
When I plan on releasing the plugin, I get all the translated files here : https://fr.crowdin.com/project/wirelessredstone
And I put them in the resources folder.

That's why you got all those nullpointerexceptions : the files were not there.
For this version (1.9.5 it seems) I simply want to put these files manually in the .jar before putting on bukkitdev.
For next one, I plan on making the plugin download them directly from crowdin.

**Comment by licks92** [comment]
In the end, I've read it all. Many typo fixes (I come from C++, I'm not really used to Java typo but that's okay i'll try to get used to it), and the fix for 1.8 seems working.

So... welcome in the team!

**Comment by Bart-0110** [comment]
For some reason java doesn't like it when a varible is initialized from another class.
Something that I noticed while testing

In wirelessstring.java we have
public String backupDone;

If you give it a value in wirelessxmlstring.java and access it from another class it will give a nullpointetexception. But if you give it a value on the construct of wirelessstring.java it won't give a nullpointerexception.

I'll show it tomorrow

**Comment by Bart-0110** [comment]
I don't know why it happens tho

**Comment by licks92** [comment]
Maybe because the file wasn't read don't you think?
I've seen that when i coded it months ago.

XML won't warn you if he didn't find anything at the node you asked, it
will simply fill the variable with null... which is dumb I do agree.

2015-06-14 23:11 GMT+02:00 Bart-0110 <notifications@github.com>:

**Comment by licks92** [comment]
Trying your version right now :)

**Comment by licks92** [comment]
It's working smooth. Thank you for your work!
I've been kind of bored by coding this plugin since months, but you gave me back the desire to code and thank you for that.
The next release will be soon I suppose :)

**Comment by licks92** [comment]
Hmmm info working, list working. the database seems to be working...

**Pull request merged**